# Import Data

In [ ]:
import pandas as pd

In [ ]:
#File dataset
file_id = "1hVsQ7GLg2eZoiRZtisDbuAP9qL50h_2Q"
url = f"https://drive.google.com/uc?id={file_id}"

#Baca CSV dari Drive
student_exam_scores = pd.read_csv(url)


In [ ]:
student_exam_scores.head()

,student_id,hours_studied,sleep_hours,attendance_percent,previous_scores,exam_score
0,S001,8.0,8.8,72.1,45,30.2
1,S002,1.3,8.6,60.7,55,25.0
2,S003,4.0,8.2,73.7,86,35.8
3,S004,3.5,4.8,95.1,66,34.0
4,S005,9.1,6.4,89.8,71,40.3


# Pemeriksaan Data

In [ ]:
#Melihat tipe data, apakah terdapat tipe data yang tidak sesuai

student_exam_scores.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   student_id          200 non-null    object 
 1   hours_studied       200 non-null    float64
 2   sleep_hours         200 non-null    float64
 3   attendance_percent  200 non-null    float64
 4   previous_scores     200 non-null    int64  
 5   exam_score          200 non-null    float64
dtypes: float64(4), int64(1), object(1)
memory usage: 9.5+ KB


# Pembersihan Data

In [ ]:
#Menghapus kolom yang tidak diperlukan
student_exam_scores = student_exam_scores.dropna(axis=1, how='all')

In [ ]:
#Melihat apakah terdapat missing values  atau data kosong (null)

student_exam_scores.isnull().sum()

,0
student_id,0
hours_studied,0
sleep_hours,0
attendance_percent,0
previous_scores,0
exam_score,0


In [ ]:
#Melihat apakah terdapat data duplikat

student_exam_scores.duplicated().sum()

np.int64(0)

In [ ]:
student_exam_scores.describe()

,hours_studied,sleep_hours,attendance_percent,previous_scores,exam_score
count,200.000000,200.000000,200.000000,200.000000,200.000000
mean,6.325500,6.622000,74.830000,66.800000,33.955000
std,3.227317,1.497138,14.249905,15.663869,6.789548
min,1.000000,4.000000,50.300000,40.000000,17.100000
25%,3.500000,5.300000,62.200000,54.000000,29.500000
50%,6.150000,6.700000,75.250000,67.500000,34.050000
75%,9.000000,8.025000,87.425000,80.000000,38.750000
max,12.000000,9.000000,100.000000,95.000000,51.300000


In [ ]:
num_cols = student_exam_scores.select_dtypes(include=["float64","int64"]).columns

for col in num_cols:
    Q1 = student_exam_scores[col].quantile(0.25)
    Q3 = student_exam_scores[col].quantile(0.75)
    IQR = Q3 - Q1

    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    outliers = student_exam_scores[(student_exam_scores[col] < lower_bound) | (student_exam_scores[col] > upper_bound)]

    print(f"Kolom: {col}, Jumlah Outlier: {len(outliers)}")

Kolom: hours_studied, Jumlah Outlier: 0
Kolom: sleep_hours, Jumlah Outlier: 0
Kolom: attendance_percent, Jumlah Outlier: 0
Kolom: previous_scores, Jumlah Outlier: 0
Kolom: exam_score, Jumlah Outlier: 0


# Export Data ke File CSV

In [ ]:
from google.colab import files

# Simpan ke CSV
student_exam_scores.to_csv("student_exam_scores_clean.csv", index=False, sep=";", decimal=",")

# Download
files.download("student_exam_scores_clean.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# Kesimpulan

Dalam proses pengolahan dataset student_exam_scores, beberapa langkah data cleaning dan pengecekan kualitas data telah dilakukan agar data lebih konsisten, representatif, dan siap dipakai untuk analisis lebih lanjut.

**1. Pengecekan ringkasan statistik dengan** `.describe()`

Ringkasan deskriptif digunakan untuk memahami sebaran data pada setiap variabel (hours_studied, sleep_hours, attendance_percent, previous_scores, dan exam_score). Dari sini dapat diketahui nilai minimum, maksimum, kuartil, serta rata-ratanya.

**2. Pengecekan distribusi dan outlier menggunakan metode IQR**

Outlier dicek dengan menghitung batas bawah (Q1 – 1.5 × IQR) dan batas atas (Q3 + 1.5 × IQR). Hasil awal menunjukkan bahwa pada variabel exam_score terdapat kemungkinan nilai ekstrem (misalnya nilai 17.1 yang jauh di bawah Q1). Variabel lain masih berada dalam rentang yang wajar.

**3. Identifikasi fungsi outlier dalam analisis**

Outlier penting untuk diperhatikan karena bisa menunjukkan adanya kesalahan input data, kondisi khusus yang berbeda dari populasi umum, atau informasi penting yang perlu dianalisis terpisah. Dalam konteks penelitian, outlier bisa mempengaruhi model statistik maupun machine learning jika tidak ditangani dengan tepat.

**4. Penanganan awal outlier**

Strategi umum yang dapat digunakan meliputi:

*   Menghapus outlier jika terbukti sebagai kesalahan pencatatan.
*   Mengganti (imputasi) dengan median agar distribusi data tetap stabil.
*   Membiarkan outlier jika memang mencerminkan kondisi riil (misalnya siswa dengan nilai ujian sangat rendah karena faktor khusus).

**5. Ekspor data hasil cleaning ke CSV/Excel.**

Dataset hasil olahan disimpan kembali agar mudah dibagikan dan digunakan ulang.
Perbedaan locale (. vs , untuk desimal) diatasi dengan menyesuaikan opsi ekspor atau langsung menyimpan dalam format Excel (.xlsx).